spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java

```diff
@@ -184,6 +185,22 @@ public void setRequestsMetricName(String requestsMetricName) {
 		 */
 		private final Map<String, ServiceLevelAgreementBoundary[]> sla = new LinkedHashMap<>();
 
+		/**
+		 * The minimum value that this distribution summary is expected to observe.
+		 * Controls the number of buckets shipped by percentilesHistogram. Can be
+		 * specified as a long or as a Duration value (for timer meters, defaulting to ms
+		 * if no unit specified).
+		 */
+		private final Map<String, ServiceLevelAgreementBoundary> minimumExpectedValue = new LinkedHashMap<>();
```

Comment:
While `ServiceLevelAgreementBoundary` is a convenient type to deserialize this value, `minimumExpectedValue` is unconnected from SLAs conceptually.

Reply:
WDYT @snicoll?

Reply:
Agree. Which alternative do you guys recommend:
- copy ServiceLevelAgreementBoundary to a better named class with exactly the same code
- rename ServiceLevelAgreementBoundary
- other?

Reply:
ping @snicoll

Reply:
I frankly have no opinion as I am not sure I've grasped those concepts fully just yet. I've flagged for team attention to get this one more visibility. We have a call today so we should provide a smarter feedback soon.

Thanks for your patience.

Reply:
It's unfortunate that the ideal type is either a `long` or a `Duration`. Separate properties for distribution summaries (`long`) and timers (`Duration`) would offer more type-safety in the configuration and could also simplify the property descriptions. I'm not sure what those properties would be called though.

Failing that, I'd type the value as `String` and then convert that to a `long` or `Duration` using the conversion service. That should ensure the same conversion behaviour as if the properties themselves had been more strongly typed.

Reply:
I guess that ship has sailed with the `sla` property (or we can make this compromise only there). The problem by having two target types depending on the meter type is that we can't really get proper IDE support for this.

I am not sure how we could find a proper name for those and the convenience of doing it on a per meter id is quite appealing. So a `String` to long or `Duration` looks like the way to go.

@alexanderabramov are you interested to revisit the PR short term? RC1 is due soon.